test/testdrive/catalog.td

```diff
@@ -385,12 +385,18 @@ contains:unknown catalog item 'v'
 > CREATE DATABASE d
 
 # Dropping the public schema is okay, but dropping the catalog schemas is not.
-> DROP SCHEMA public
+$ postgres-execute connection=postgres://mz_system@materialized:6877/materialize
+DROP SCHEMA public CASCADE
+ALTER SYSTEM SET enable_rbac_checks TO false
```

Comment:
You dropped the public schema here which is why the `CREATE TYPE` on line 651 doesn't work anymore.

Reply:
But the `DROP SCHEMA public` was also here before in line 388, maybe it is the cascade I added. I'll try removing that.

Reply:
> catalog.td:388:1: error: executing postgres query: db error: ERROR: schema 'materialize.public' cannot be dropped without CASCADE while it contains objects: ERROR: schema 'materialize.public' cannot be dropped without CASCADE while it contains objects

Reply:
Previously this was executed as the `materialize` role which meant that it would drop the public schema from the current database, which is `d`. So previously it was equivalent to `DROP SCHEMA d.public`.

Now it's being run in the `mz_system` user's session, which has the current database set to `materialize`. So now it's equivalent to `DROP SCHEMA materialize.public`. That's why it now requires `CASCADE` when previously it didn't. 

You should be able to switch this back to running as the `materialize` role, since `materialize` owns the database `d`.